mina.netty/src/main/java/org/kaazing/mina/netty/socket/nio/DefaultNioDatagramChannelIoSessionConfig.java

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.mina.netty.socket.nio;
+
+import org.jboss.netty.channel.FixedReceiveBufferSizePredictorFactory;
+import org.jboss.netty.channel.ReceiveBufferSizePredictor;
+import org.jboss.netty.channel.ReceiveBufferSizePredictorFactory;
+import org.jboss.netty.channel.socket.DefaultDatagramChannelConfig;
+import org.jboss.netty.channel.socket.nio.NioDatagramChannelConfig;
+
+import java.net.DatagramSocket;
+import java.net.SocketException;
+
+public class DefaultNioDatagramChannelIoSessionConfig extends NioDatagramChannelIoSessionConfig {
+
+    private static final byte REUSEADDRESS_MASK = 1;
+    private static final byte RECEIVEBUFFERSIZE_MASK = 2;
+    private static final byte SENDBUFFERSIZE_MASK = 4;
+    private static final byte TRAFFICCLASS_MASK = 8;
+
+    // bit mask to tell if a field has been set or not
+    private byte fieldSetMask;
+
+    private boolean reuseAddress;
+    private int receiveBufferSize;
+    private int sendBufferSize;
+    private int trafficClass;
+
+    public DefaultNioDatagramChannelIoSessionConfig() {
+        super(new DefaultNioDatagramChannelConfig(newDatagramSocket()));
+        // We could make this conditional (if !ENABLE_BUFFER_SIZE) to allow users to turn on variable buffer size via
```

Comment:
TrafficShapingFilter only works with tcp so this comment is not relevant for UDP. Our decision about fixed or variable receive buffer size should not be driven by TrafficShapingFilter considerations for UDP.